### ISSUE TITLE
Pretrained weights and position embedding at 224x224

### ISSUE THREAD

**Issue by rwightman** [google-research/vision_transformer#6]
I've been fiddling with the models a bit, massaging the weights into my PyTorch impl. I have the base 384x384 models working well, but generating the params for the 224x224, the resulting output has low validation accuracy (it was in the 76s top-1 when I killed it).

Is that expected? Is the pos embedding interpolation from 24x24 grid only intended as a starting point for transfer learning and not expected to provide good results as is? By comparison the base 384x384 16x16 patch is validating at 84.2 and the 384x384 32x32 patch at 81.7

**Comment by akolesnikoff** [comment]
Yes, when changing the input resolution from 384x384 to 224x224 there is indeed quite significant drop in performance. For the released ViT-B/16 model finetuned on ImageNet I've observed 9.5% top-1 accuracy drop on the ImageNet validation split.

I will look into training and releasing models that expect 224x224 inputs in the coming weeks.

**Comment by rwightman** [comment]
@akolesnikoff Thanks for the info, your numbers match mine. Do you want me to leave this open until 224x224 native models are released?

**Comment by akolesnikoff** [comment]
yes, let's keep it open until new models arrive.

**Comment by akolesnikoff** [comment]
We have added a couple of models finetuned at 224x224 input resolution, see update in README for details: https://github.com/google-research/vision_transformer#available-models

**Comment by rwightman** [comment]
@akolesnikoff thanks, they look good. BTW, the 224 weights validate a bit better with a crop (zoom), I use 0.9 as it worked well for my own 224x224 ImageNet-1k training. I get 81.78 and 83.07 with that crop in PyTorch on your new weights, likely a bit higher with same input pipeline as training.

**Comment by dribnet** [comment]
Thanks for these. Should we also add an entry in KNOWN_MODELS? It's not clear to me how to use these.

```python
        model = ???
        self.VisionTransformer = models.KNOWN_MODELS[model].partial(num_classes=1000)
        self.params = checkpoint.load('ViT-B_16-224.npz')
```

**Comment by dribnet** [comment]
FWIW: my ilsvrc2012 validation scores for these *-224 models using the inference code from the colab.

|     |      ViT-B_16-224      |  ViT-L_16-224 |
|----------|-------------:|------:|
| Top-1 |  80.6% | 82.3%  |
| Top-5 |  95.6%  |  96.2% |

IIUC (and if I didn't screw this up) the L_16-224 drop would only be 3% relative to the published "Ours-I21K (ViT-L/16)" figure of 85.3 and is less that I would have expected from the above discussion.

**Comment by akolesnikoff** [comment]
~10% drop happens when the model fine-tuned with 384x384 resolution is directly applied at 224x224 images.

However, when comparing two different models properly finetuned at 384x384 and 224x224, the drop is ~3%.

Does this resolve your confusion?
